liquidator/index.js

```diff
@@ -6,18 +6,30 @@ const { delay } = require("../financial-templates-lib/delay");
 const { Logger } = require("../financial-templates-lib/Logger");
 
 // JS libs
-const { Liquidator } = require("./Liquidator");
+const { Liquidator } = require("./liquidator");
 const { GasEstimator } = require("../financial-templates-lib/GasEstimator");
 const { ExpiringMultiPartyClient } = require("../financial-templates-lib/ExpiringMultiPartyClient");
 
 // Truffle contracts
 const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
 
 // TODO: Figure out a good way to run this script, maybe with a wrapper shell script.
-// Currently, you can run it with `truffle exec ../liquidator/liquidator.js --address=<address> --price=<price>` *from the core
+// Currently, you can run it with `truffle exec ../liquidator/index.js --address=<address> --price=<price>` *from the core
 // directory*.
 
-async function run() {
+/**
+ * @notice Runs the main liquidator indefinitely, unless the caller passes a test configuration object,
+ * in which case the script will exit after one iteration and return true if no errors were thrown.
+ * @param testConfig { address, price } Substitutes for command line flags that would be passed in normally.
+ * @return None or throws an Error.
+ */
+async function run(testConfig) {
```

Comment:
nit: slight preference against overwriting argv. Not super opinionated though. One alternative structure you might consider:
```
async function run(price, address, shouldLoop) {
```

And then you could move all the argv parsing stuff down to the Poll method since you can't really test all of the minimist parsing in tests anyway.

I guess the one advantage of doing it the way you're doing it is that it technically allows you to explicitly test the argv checks whereas you wouldn't be able to if you put it in the Poll method.

Up to you -- just wanted to mention the alternative.

Reply:
yeah I had originally done it this way so that I could test the argv checks, but I understand if don't want to manipulate that object.

I like the idea of moving the argv parising to the Poll method and will make that change. Its a cleaner separation of concerns